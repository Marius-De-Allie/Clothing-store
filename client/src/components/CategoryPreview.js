import React from 'react';
import { Link } from 'react-router-dom';
import CategoryItem from './CategoryItem';
// styles.
import '../styles/categoryPreview.scss';

const CategoryPreview = ({ title, items, routeName }) => {

    return (
        <div className="category-preview">
            <h1 className="title">
                <Link className="cat-link" to={`/shop/${routeName}`}>{title}</Link>
            </h1>
            <div className="preview">
                {items.filter((el, index) => index < 4).map(item => 
                    <CategoryItem key={item.id} item={item} />
                )}
            </div>
        </div>
    );
};

export default CategoryPreview;