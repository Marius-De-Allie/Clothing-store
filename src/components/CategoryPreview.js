import React from 'react';
import CategoryItem from './CategoryItem';
// styles.
import '../styles/categoryPreview.scss';

const CategoryPreview = ({ title, items }) => {
    console.log(items)

    return (
        <div className="category-preview">
            <h1 className="title">{title}</h1>
            <div className="preview">
                {items.filter((el, index) => index < 4).map(item => 
                    <CategoryItem key={item.id} {...item} />
                )}
            </div>
        </div>
    );
};

export default CategoryPreview;