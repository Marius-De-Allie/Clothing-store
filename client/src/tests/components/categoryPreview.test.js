import React from 'react';
import { shallow } from 'enzyme';
import CategoryPreview from '../../components/CategoryPreview';

it('CategoryPreview component renders propery', () => {
    // Mock array to pass into CategoryPreview comp.
    const mockItems = [
        {
            id: 1,
            imageUrl: '',
            name: '', 
            price: 0
        }
    ];
    
    const wrapper = shallow(<CategoryPreview items={mockItems} />);
    expect(wrapper).toMatchSnapshot();
});