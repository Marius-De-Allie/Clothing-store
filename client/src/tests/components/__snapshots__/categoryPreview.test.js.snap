// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPreview component renders propery 1`] = `ShallowWrapper {}`;
